runtime/sak.c

```diff
@@ -34,7 +34,12 @@
 #define printf_os wprintf
 #else
 #define strncmp_os strncmp
+#ifdef __NetBSD__
+/* NOTE: See CAVEATS section in https://man.netbsd.org/ctype.3 */
+#define toupper_os(x) toupper((unsigned int)x)
```

Comment:
Given this from that manpage:
```
Some implementations of libc, such as glibc as of 2018, attempt to avoid
     the worst of the undefined behavior by defining the functions to work for
     all integer inputs representable by either unsigned char or char, and
     suppress the warning.  However, this is not an excuse for avoiding con-
     version to unsigned char: if EOF coincides with any such value, as it
     does when it is -1 on platforms with signed char, programs that pass char
     will still necessarily confuse the classification and mapping of EOF with
     the classification and mapping of some non-EOF inputs.
```

Should we just unconditionally cast to `(unsigned int)` on all platforms, including glibc?

Reply:
oooops, I meant to use `unsigned char` not `unsigned int`. Fixed.

Reply:
Thanks; looking at the glibc man page for toupper, it looks appropriate to always cast `toupper_os()`'s argument to `unsigned char`:

```
The standards require that the argument c for these functions is
       either EOF or a value that is representable in the type unsigned
       char.  If the argument c is of type char, it must be cast to
       unsigned char, as in the following example:

           char c;
           ...
           res = toupper((unsigned char) c);

       This is necessary because char may be the equivalent signed char,
       in which case a byte where the top bit is set would be sign
       extended when converting to int, yielding a value that is outside
       the range of unsigned char.
```

Reply:
Faire enough. Done